src/Container/Container.php

```diff
@@ -0,0 +1,268 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Container;
+
+use Cake\Container\Definition\DefinitionAggregate;
+use Cake\Container\Definition\DefinitionAggregateInterface;
+use Cake\Container\Definition\DefinitionInterface;
+use Cake\Container\Exception\ContainerException;
+use Cake\Container\Exception\NotFoundException;
+use Cake\Container\Inflector\InflectorAggregate;
+use Cake\Container\Inflector\InflectorAggregateInterface;
+use Cake\Container\Inflector\InflectorInterface;
+use Cake\Container\ServiceProvider\ServiceProviderAggregate;
+use Cake\Container\ServiceProvider\ServiceProviderAggregateInterface;
+use Cake\Container\ServiceProvider\ServiceProviderInterface;
+use Psr\Container\ContainerInterface;
+
+class Container implements DefinitionContainerInterface
```

Comment:
Would we switch the base class of `Cake\Core\Container` in a future 5.x release or leave that to 6.x?

Reply:
That would be a 6.x only move since major changes have already happened here + more major changes are going to happen according to ADmad's christmas wishlist bellow.